compiler/circle-opselector/driver/Driver.cpp

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "OpSelector.h"
+
+#include <foder/FileLoader.h>
+
+#include <luci/Importer.h>
+#include <luci/CircleExporter.h>
+#include <luci/CircleFileExpContract.h>
+#include <luci/Import/CircleReader.h>
+
+#include <luci/Profile/CircleNodeID.h>
+
+#include <arser/arser.h>
+#include <vconone/vconone.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+void print_version(void)
+{
+  std::cout << "circle-opselector version " << vconone::get_string() << std::endl;
+  std::cout << vconone::get_copyright() << std::endl;
+}
+
+bool check_input(const std::string str)
+{
+  bool has_hyphen = false;
+
+  if (str.empty())
+    return false;
+  if (str.at(0) == '-' || str[str.size() - 1] == '-')
+  {
+    std::cerr << "Invalid input." << std::endl;
+    return false;
+  }
+
+  for (char c : str)
+  {
+    if (isdigit(c))
+      continue;
+    else if (has_hyphen && c == '-') // when user enter '-' more than 2.
+    {
+      std::cerr << "Too many '-' in str." << std::endl;
+      return false;
+    }
+    else if (c == '-')
+      has_hyphen = true;
+    else // when user enter not allowed character, print alert msg.
+    {
+      std::cerr << "To select operator by id, please use these args: [0-9], '-', ','" << std::endl;
+      return false;
+    }
+  }
+  return true;
+}
+
+void split_id_input(const std::string &str, std::vector<int> &by_id)
+{
+  std::istringstream ss;
+  ss.str(str);
+  std::string str_buf;
+
+  while (getline(ss, str_buf, ','))
+  {
+    if (str_buf.length() && check_input(str_buf)) // input validation
```

Comment:
Thank you for your comment!

We already handle that cases with the `check_input` method.
> - 10,,,,11
> - - empty string is permitted and continue next token

This is handled by `if(str.empty())` block of `check_input`.
> - 11-,12
> - - Ignore 11- (because it ends with -) and continue with next token. It will select 12 only.

This is handled by `if (str.at(0) == '-' || str[last_idx] == '-')` block of `check_input`.

Reply:
@dongyooncho I know. I am mentioning it continue to parse with next token after `check_input` returns `false`. I prefer to exit when `check_input` fails.